### DISCUSSION TITLE
How do I configure meson for embedded systems?

### DISCUSSION THREAD

**Discussion by NachoIDAN** [mesonbuild/meson#10000] [Q&A]
I'm a newbie with meson. 
We are using a compiler for embedded systems (tasking compiler), which does not generate executables but rather hex files (to flash into an embedded system).
Does anyone has experience with this to assist me in configuring meson.
We have been using a "home-brewed" system, but I would like to switch to meson or scons.
Does meson support the use of option files?
The command line we used (very long time ago), was so long that it was required to put the options in a file and use a -f xx.opt (-f: read options from file), because the command line was over 1K long (file is actually 5K long).

**Comment by xclaesse** [comment]
Yes you can write all your options into a config file: https://mesonbuild.com/Machine-files.html#project-specific-options.

**Comment by eli-schwartz** [comment]
Meson has internal support for compiler response files, taking advantage of the fact that it natively knows how to build them and what format they need to be.

This depends on the compiler class inside meson knowing that a particular compiler supports the concept.

Since you're generating hex files rather than executables, are you using custom_target? I'm not sure what a response/option file would look like there. We'd probably need to add a new feature in order to support them directly...

Something like perhaps adding a `response_file: 'foo.opt'` which would support the usual `@VAR@` replacements supported in the command array? And then you could define a direct command `[..., '-f', '@RESPONSE_FILE@']`

**Comment by jpakkane** [comment]
There is a [sample project](https://github.com/mesonbuild/meson/tree/master/test%20cases/common/51%20run%20target) you can steal from in Meson's test suite. It even shows you how to set up an upload target so all you need to do is `ninja upload` to build and flash the firmware.
